test/integration/infrastructure/infrastructure_test.go

```diff
@@ -250,6 +253,102 @@ var _ = Describe("Infrastructure tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+
+	Context("with invalid credentials", func() {
+		It("should fail creation but succeed deletion", func() {
+			providerConfig := newProviderConfig(awsv1alpha1.VPC{
+				CIDR: pointer.StringPtr(vpcCIDR),
+			})
+
+			namespaceName, err := generateNamespaceName()
+			Expect(err).NotTo(HaveOccurred())
+
+			var (
+				namespace *corev1.Namespace
+				cluster   *extensionsv1alpha1.Cluster
+				infra     *extensionsv1alpha1.Infrastructure
+			)
+
+			framework.AddCleanupAction(func() {
+				By("cleaning up namespace and cluster")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+				Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+			})
+
+			defer func() {
```

Comment:
`defer` won't be called if the test execution is interrupted by SIGINT/SIGTERM. You could use `AddCleanupAction` instead to make sure that the test won't leak any infra if it is interrupted.

Reply:
Well, in this test we can be sure, that no infra is leaked as we pass invalid credentials.

As for the other test cases, we should actually also move the deletion logic to a `defer` and only do actual cleanup on AWS in a `CleanupAction` (see https://github.com/gardener/terraformer/pull/57 for more background on that).
However, I didn't tackle this topic in this PR.

Reply:
> Well, in this test we can be sure, that no infra is leaked as we pass invalid credentials.

Ah, yep, I missed this point.